sources/web/datalab/polymer/modules/drive-file-manager/drive-file-manager.ts

```diff
@@ -116,27 +126,33 @@ class DriveFileManager extends BaseFileManager {
       // TODO - create the real path to this object, or figure out
       // a better way to handle not having the full path in the breadcrumbs
       const fileId = path;  // We assume the entire path is one fileId
-      const datalabFile: DriveFile = new DriveFile({
-        id: new DatalabFileId(fileId, FileManagerType.DRIVE),
-      } as DatalabFile);
+      const datalabFile: DriveFile = new DriveFile(
+        new DatalabFileId(fileId, FileManagerType.DRIVE),
+        path,
+        DatalabFileType.DIRECTORY,
+      );
       return [datalabFile];
     }
   }
 
   protected _fromUpstreamFile(file: gapi.client.drive.File) {
-    const datalabFile: DriveFile = new DriveFile({
-      icon: file.mimeType === DriveFileManager._directoryMimeType ?
-                              file.iconLink : 'editor:insert-drive-file',
-      id: new DatalabFileId(file.id, FileManagerType.DRIVE),
-      name: file.name,
-      type: file.mimeType === DriveFileManager._directoryMimeType ?
+    const driveFile = new DriveFile(
+      new DatalabFileId(file.id, FileManagerType.DRIVE),
+      file.name,
+      file.mimeType === DriveFileManager._directoryMimeType ?
                               DatalabFileType.DIRECTORY :
                               DatalabFileType.FILE,
-    } as DatalabFile);
-    if (datalabFile.name.endsWith('.ipynb')) {
-      datalabFile.type = DatalabFileType.NOTEBOOK;
+      file.mimeType === DriveFileManager._directoryMimeType ?
+                              file.iconLink : 'editor:insert-drive-file',
+    );
+    if (driveFile.type === DatalabFileType.FILE && driveFile.name.endsWith('.ipynb')) {
+      driveFile.type = DatalabFileType.NOTEBOOK;
+    }
+    driveFile.lastModified = new Date(file.modifiedTime).toDateString();
```

Comment:
Using toDateString() here shows only the date and not the time (yes, I know the Drive UI doesn't show mod time, but we could). Maybe you could do it like <code>ls -l</code> does, showing the month/day/time when less than six months old, and showing the month/day/year when more than six months old. Or maybe use Date.toLocaleString().

Reply:
Switched to using `toLocaleString`.